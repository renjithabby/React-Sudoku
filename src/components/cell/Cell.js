import React from 'react';
import PropTypes from 'prop-types';

import './style.css';


const Cell =  ({id, invalid, value, prefilled, handleCellClick }) => {
    return (<div onClick={handleCellClick}  id={id} className= {`cell-container cell-container--${invalid?'invalid':'valid'} ${prefilled?'cell-container--prefilled':''}`}>
        <span className="cell-value">{value?value:''}</span> 
    </div>)
};

export default Cell;

Cell.propTypes= {
    id : PropTypes.number.isRequired,
    invalid : PropTypes.bool,
    value   : PropTypes.number,
    prefilled : PropTypes.bool
}

Cell.defaultProps= {
    invalid : false,
    value   : 0,
    prefilled : false
}
