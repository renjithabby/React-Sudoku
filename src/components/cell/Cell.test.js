import React from 'react';
import {configure, shallow} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import toJson from 'enzyme-to-json';
import Cell from './Cell';

configure({adapter : new Adapter()});

describe('<Cell />', () => {

    it('It matches the snapshot with default values', () => {
        const tree = shallow(<Cell id={0}/>);
        expect(toJson(tree)).toMatchSnapshot();
    });

    it('It matches the snapshot with props "prefilled" & "invalid" set to "true"', () => {
        const tree = shallow(<Cell id ={0} value={6} prefilled={true} invalid={true}/>);
        expect(toJson(tree)).toMatchSnapshot();
    });

})