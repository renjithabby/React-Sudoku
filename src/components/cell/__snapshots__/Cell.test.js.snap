// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Cell /> It matches the snapshot with default values 1`] = `
<div
  className="cell-container cell-container--valid "
  id={0}
>
  <span
    className="cell-value"
  />
</div>
`;

exports[`<Cell /> It matches the snapshot with props "prefilled" & "invalid" set to "true" 1`] = `
<div
  className="cell-container cell-container--invalid cell-container--prefilled"
  id={0}
>
  <span
    className="cell-value"
  >
    6
  </span>
</div>
`;
